### DISCUSSION TITLE
How to test next function in beforeRouteEnter()

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [vuejs/test-utils#2509] [Q&A]
Hi,
I have the following code:
```
beforeRouteEnter(to, from, next) {
    next((vm: any) => {
      if (to.name !== from.name) {
        vm.$store.commit('randomFunction');
      }
    });
  },
```
How would I create a test that checks $store.commit was called with 'randomFunction'?
I've read you can test before RouteEnter with this `MyComponent.beforeRouteEnter.call(wrapper.vm, undefined, undefined, next)` but the issue with this is I have to define the next function in my test class which would mean it won't detect changes in the .vue class and correctly signal failures if the logic has changed.

**Comment by cexbrayat** [comment]
Hi @steven-twerdochlib 
You should check out https://github.com/posva/vue-router-mock and trigger a navigation to the component you want to test, so that beforeRouteEnter run.
